packages/ember-metal/lib/streams/key-stream.js

```diff
@@ -25,6 +25,12 @@ let KeyStream = BasicStream.extend({
 
   compute() {
     var object = this.sourceDep.getValue();
+    const type = typeof object;
+
+    if (type === 'boolean' || type === 'string') {
```

Comment:
I don't think this is correct, we should take this branch if a property access is not possible. It might be easier to express the negation.

Property access are allowed iff
* object !== null && type === 'object' || type === 'function'